qiskit_experiments/base_experiment.py

```diff
@@ -98,19 +99,34 @@ def run(
 
         Returns:
             The experiment data object.
+
+        Raises:
+            QiskitError: if experiment is run with an incompatible existing
+                         ExperimentData container.
         """
-        # Create new experiment data
         if experiment_data is None:
-            experiment_data = self.__experiment_data__(self, backend=backend)
-
-        # Generate and transpile circuits
-        circuits = transpile(self.circuits(backend), backend, **self.transpile_options.__dict__)
+            # Create new experiment data
+            experiment_data = self.__experiment_data__(experiment=self, backend=backend)
+        else:
+            # Validate experiment is compatible with existing data container
+            metadata = experiment_data.metadata()
+            if metadata.get("experiment_data") != self._type:
+                raise QiskitError(
+                    "Existing ExperimentData contains data from a different experiment."
+                )
+            if metadata.get("physical_qubits") != list(self.physical_qubits):
+                raise QiskitError(
+                    "Existing ExperimentData contains data for a different set of physical qubits."
+                )
```

Comment:
Consider adding checks on some run options as different measurement levels could cause data processing issues.

Reply:
I think this one is a bit more experiment specific. I could also imagine some experiments work with multiple measurement levels if their analysis can check the meas level and use appropriate processor. Worth discussing more for a possible follow up PR

Reply:
Good point.